### PR TITLE
Add an indexstore version API

### PR DESCRIPTION
**Pull request by bnbarham**
This allows skipping tests if the corresponding libIndexStore version is too old.

### CODE REVIEW

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
Let me know if there's an easier way to thread that API through 😅. Ended up touching a fair bit more than I thought I would.

**Review by benlangmuir — COMMENTED** [review]
I think you should expose this on `IndexStoreLibrary` instead, and expose the format version the same way while you're at it. I think putting it on that object is preferable, because this is something you would want to know *before* creating an IndexStore**DB** using the given library if you were going to e.g. multiplex across format versions.

I think this should be doable by making `indexstoredb_store_version` take a `indexstoredb_indexstore_library_t` instead of `indexstoredb_index_t`.  In C++ world that should give you access to a `IndexStoreLibrary` that has the API functions set.

API-wise, I wonder what your thoughts are on returning `(Int, Int)` for the version and splitting it back into major/minor?

**Comment by bnbarham** [comment]
> I think you should expose this on `IndexStoreLibrary` instead, and expose the format version the same way while you're at it. I think putting it on that object is preferable, because this is something you would want to know _before_ creating an IndexStore**DB** using the given library if you were going to e.g. multiplex across format versions.
> 
> I think this should be doable by making `indexstoredb_store_version` take a `indexstoredb_indexstore_library_t` instead of `indexstoredb_index_t`. In C++ world that should give you access to an `indexstore::IndexStoreLibrary` that has the API functions set.

Thanks. That's a lot better than threading it all the way through `IndexSystem`.

> API-wise, I wonder what your thoughts are on returning `(Int, Int)` for the version and splitting it back into major/minor?

Sure, I ended up making it a type 👍.

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please test

File: Tests/IndexStoreDBTests/IndexTests.swift [hunk 1]
```diff
@@ -661,11 +661,12 @@ final class IndexTests: XCTestCase {
 
   func testConcepts() throws {
     guard let ws = try staticTibsTestWorkspace(name: "CxxLangFeatures") else { return }
+    try XCTSkipIf(ws.libIndexStore.version <= IndexStoreLibrary.Version(major: 0, minor: 14))
```
[the remark this answers is not in the document]
Comment (bnbarham) [reply Tests/IndexStoreDBTests/IndexTests.swift]: Indeed, thanks

File: Sources/IndexStoreDB/IndexStoreDB.swift [hunk 2]
```diff
@@ -390,6 +390,35 @@ public protocol IndexStoreLibraryProvider {
...
+
+    public init(encoded: Int) {
+      self.init(major: encoded / 10000, minor: encoded % 10000)
+    }
+
+    public static func < (lhs: Version, rhs: Version) -> Bool {
+      if lhs.major != rhs.major {
+        return lhs.major < rhs.major
+      }
+      return lhs.minor < rhs.minor
```
[the remark this answers is not in the document]
Comment (bnbarham) [reply Sources/IndexStoreDB/IndexStoreDB.swift]: Very pretty, changed.

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**